src/btree/bt_compact.c

```diff
@@ -259,6 +258,8 @@ __compact_walk_internal(WT_SESSION_IMPL *session, WT_REF *parent)
             WT_ERR(__compact_page(session, ref, &skipp));
             if (!skipp)
                 overall_progress = true;
+            else if (session->compact_state == WT_COMPACT_EXITING)
```

Comment:
Since we don't really want to use the block manager here and the `WT_BLOCK` structure, this is the only way @Sean04 and I found to tell compaction to stop.
If you have any other ideas @pmacko86 @kommiharibabu, please let us know.

Reply:
I went with `else if` as I am assuming `skip` has to be `true`. That's the only case where `WT_COMPACT_EXITING` is set so far. However, I agree it can look odd, especially if we start setting `WT_COMPACT_EXITING` under different conditions.

Reply:
Fixed in 3db2dec7f21469fee8688e67dd8a36d454998958 where I also moved the check before the one on `skip`.